### ISSUE TITLE
TimeSpan.Humanize() really should round

### ISSUE THREAD

**Issue by mqudsi** [Humanizr/Humanizer#648]
I'll start this off by saying I've read a lot of the previous issues regarding `TimeSpan` humanization and have read the discussions on precision, maxUnit, minUnit, etc.

That said, I agree with the premise that when dealing with larger units, smaller units are not as important and can be disregarded by default; i.e. 4 hours and 3 minutes is four hours.

However, if we are discarding smaller units, then it is incorrect to truncate times via integer division.

Imagine it like this: you want to tell someone as accurately as possible that there are 3 hours and 59 minutes remaining, but in their language, the smallest unit of time available is `hour`. What is more correct: to say that there are 3 hours remaining (expressing only the hour unit and dropping the rest) or to round up and say there are 4 hours remaining?

_I'm actually not sure why the default behavior of `TimeSpan.Humanize()` isn't more like `DateTime.Humanize()` which uses the `0.75` precision by default (4:00 counting down through 3:45 would display four hours, 3:45 onward would display 3 hours, 45 minutes), which I feel makes sense for both. However, that's a different discussion and I'll defer to the wisdom behind the initial selection on that matter._

**Comment by MehdiK** [comment]
Thanks for the comment and suggestion. 

To be perfectly honest, there was no wisdom behind this 😜  The initial implementation didn't have precision. What you see now is the result of a few changes and additions over time and it never occurred to me to keep it consistent with `DateTime.Humanize` which would've been the logical approach. Good news is there are stablished patterns like `Configurators` and `Strategy` in the codebase to deal with this. Did you want to send a PR?

**Comment by alexb5dh** [comment]
Stumbled upon this for similar reason.
Issue seems to be frozen, but I just wanted to ask if there's any simple code to achieve similar 0.75 precision humanization for `TimeSpan`. It seems that strategies are not applicable in this case.

**Comment by gavinltomra** [comment]
Yeah, I've just run into this with "1 day, 23 hours" being shown as "1 day" rather than "2 days" as expected with the default precision=1.

I also had "1 day, 12 hours" being shown as "1 day" which is more debatable; tempted to suggest that "1.5 days" might be the best output there.

**Comment by clairernovotny** [comment]
Thank you for the thoughtful report and examples, and to everyone who followed up here.

We re-checked this against current `main`. Changing the default `TimeSpan.Humanize()` behavior from truncation to rounding would change established results—for example, a one-part `1 day, 23 hours` value would move from `1 day` to `2 days`.

PR #1865 has now merged to `main` for the next release. It adds `Configurator.TimeSpanHumanizeStrategy`, `ITimeSpanHumanizeStrategy`, and the reusable `DefaultTimeSpanHumanizeStrategy`, so applications can opt into their preferred rounding/decomposition policy while retaining Humanizer’s culture, precision, bounds, separator, word, and symbol pipeline.

With that extensibility available, we do not plan to change the compatibility-preserving default, so I’m closing this as superseded/not planned. If a broadly reusable built-in opt-in rounding strategy is still desired, a focused proposal should specify midpoint behavior and interactions with `precision`, `maxUnit`, and `minUnit`. Thank you again for helping shape the extensibility point.
